twitter/rate_limits.go

```diff
@@ -0,0 +1,60 @@
+package twitter
+
+import (
+	"net/http"
+
+	"github.com/dghubble/sling"
+)
+
+// RateLimitService provides methods for accessing Twitter rate limits
+// API endpoints.
+type RateLimitService struct {
+	sling *sling.Sling
+}
+
+// newRateLimitService returns a new RateLimitService.
+func newRateLimitService(sling *sling.Sling) *RateLimitService {
+	return &RateLimitService{
+		sling: sling.Path("application/"),
+	}
+}
+
+// RateLimitStatusResponse response of rate limit status
+type RateLimitStatusResponse struct {
+	RateLimitContext *RateLimitContext   `json:"rate_limit_context,omitempty"`
+	Resources        *RateLimitResources `json:"resources,omitempty"`
+}
+
+// RateLimitContext contains auth context
+type RateLimitContext struct {
+	AccessToken string `json:"access_token,omitempty"`
+}
+
+// RateLimitResources contains all limit status data for endpoints group by resources
+type RateLimitResources struct {
+	Users    map[string]*RateLimitResource `json:"users,omitempty"`
+	Statuses map[string]*RateLimitResource `json:"statuses,omitempty"`
+	Help     map[string]*RateLimitResource `json:"help,omitempty"`
+	Search   map[string]*RateLimitResource `json:"search,omitempty"`
+}
+
+// RateLimitResource contains limit status data for a single endpoint
+type RateLimitResource struct {
+	Limit     int `json:"limit,omitempty"`
+	Remaining int `json:"remaining,omitempty"`
+	Reset     int `json:"reset,omitempty"`
+}
+
+// RateLimitStatusParams are the parameters for RateLimitService.Status.
+type RateLimitStatusParams struct {
+	Resources string `url:"resources,omitempty"`
```

Comment:
If the endpoint takes a list of resource kinds, the API should take a slice of those resource kinds. What if a user wanted to construct that list in a higher level function and append items to it conditionally? Let's not assume its one comma separated string please